lib/txbr/metadata.rb

```diff
@@ -0,0 +1,30 @@
+module Txbr
+  class Metadata
+    ASSIGNMENTS = %w(project_slug resource_slug prefix)
+
+    attr_reader :project_slug, :resource_slug, :prefix
+
+    def initialize(options = {})
+      @project_slug = options.fetch('project_slug')
+      @resource_slug = options.fetch('resource_slug')
+      @prefix = options.fetch('prefix')
+    end
+
+    def ==(other)
```

Comment:
Generally, I'd rather not monkey patch something not defined by us, and opt by something like the `eql?` below.

Reply:
(But if there's strong preference I'm ok with it!)

Reply:
Yeah... I hesitated a little bit, but from what I've been able to determine, redefining `==` is actually fairly common. I ran across [this SO answer](https://stackoverflow.com/a/7157051) that describes the differences between the various equality operators/functions in Ruby. The `eql?` method is for when two objects have the same hash value, and `equals?` is for when they are literally the same object (i.e. they inhabit the same address in memory). We need both `==` and `eql?` because 1) the code needs to compare two instances (and uses `==`, which I think makes sense from a usability standpoint), and 2) because the code uses these guys as `Hash` keys.